packages/string/src/constants.ts

```
@@ -105,13 +111,7 @@ export const RE_VALID_ANSI_PAIRS = /\u001B\[(\d+(?:;\d+)*)?m[^\u001B]*(?:\u001B\
  * Matches sequences like '\u001B]8;;https://example.com\u0007text\u001B]8;;\u0007'
  */
 // eslint-disable-next-line regexp/no-control-character, no-control-regex
-export const RE_VALID_HYPERLINKS = /\u001B\]8;;[^\u0007]*\u0007[^\u001B]*(?:\u001B\]8;;\u0007|$)/g;
-
-/**
- * Regular expression for ANSI link end sequences
- */
-// eslint-disable-next-line no-control-regex,regexp/no-control-character
-export const RE_ANSI_LINK_END = /\u001B\]8;;\u0007/y;
+export const RE_VALID_HYPERLINKS = /(?<=\u001B\]8;[^\u0007\u001B]*(?:\u0007|\u001B\\))(.*?)(?=\u001B\]8;;(?:\u0007|\u001B\\))/g;
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)